homeassistant/components/modbus/validators.py

```diff
@@ -130,20 +162,47 @@ def struct_validator(config: dict[str, Any]) -> dict[str, Any]:
         if not swap_type_validator:
             error = f"{name}: `{CONF_SWAP}:{swap_type}` cannot be combined with `{CONF_DATA_TYPE}: {data_type}`"
             raise vol.Invalid(error)
+    if register_size_bytes == 4 and not validator.register_size_4bytes:
+        error = f"{name}: `{CONF_REGISTER_SIZE_BYTES}: {register_size_bytes}` cannot be specified with `{CONF_DATA_TYPE}: {data_type}`"
+        raise vol.Invalid(error)
+    if register_size_bytes == 8 and not validator.register_size_8bytes:
+        error = f"{name}: `{CONF_REGISTER_SIZE_BYTES}: {register_size_bytes}` cannot be specified with `{CONF_DATA_TYPE}: {data_type}`"
+        raise vol.Invalid(error)
+    if register_size_bytes == 0 or register_size_bytes % 2 != 0:
+        error = f"{name}: Zero or odd numbers are not valid register sizes."
+        raise vol.Invalid(error)
+    if register_size_bytes not in (2, 4, 8):
+        error = f"{name}: `{CONF_REGISTER_SIZE_BYTES}:{register_size_bytes}` is out of range. Accepted values are 2, 4 or 8 bytes."
```

Comment:
"is out of range" is a bit strange, use something "is not valid, only 2, 4 or 8 are valid"

Reply:
changed